lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx

```diff
@@ -103,6 +136,113 @@ function LoadingOption({
   );
 }
 
+type LoadingErrorProps = {
+  entity: FiltersEntity;
+  retry: () => void;
+};
+
+/**
+ * Indicates an error occurred while loading filters, and presents a CTA to retry.
+ */
+function LoadingError({ entity, retry }: LoadingErrorProps) {
+  return (
+    <div
+      className="flex gap-2 items-center py-2 px-4 mb-1"
+      data-testid={`${entity}-error`}
+      // Make this element "focusable" so that clicking on it does not cause
+      // the listbox containing it to be closed
```

Comment:
This prevents the listbox to close if you click anywhere in the error message which is not the button.

TL;DR;

When clicking a non-focusable item (like a `div` without `tabIndex` explicitly set), the previously focused item looses focus, dispatching a focusout event, but the event's `relatedTarget` is null in that case, causing `useFocusAway` to close the listbox.

Basically, the workaround we used for the button and this, are solutions for two different problems which have the same effect.

These two things made it even harder for me to originally isolate the root cause because I thought I was dealing with one problem, not two.